src/builtins.rs

```diff
@@ -99,19 +100,24 @@ lazy_static! {
                     let Some(params) = parameters else { return; };
                     // Get the input and annotate it with a pointer type
                     let input = flatten_expression_list(params);
-                    let Some(input) = input.first()  else { return; };
+                    let actual_input = extract_actual_parameter(input.first().expect("Exactly one parameter required"));
```

Comment:
At first glance this expect makes it sound like if I have `REF()` the compiler will panic. However, we have a validation for the correct argument count and since this is called in the codegen we will never panic here. Can you perhaps rephrase this into `.expect("must exist; covered by validation")` or something similar to that?